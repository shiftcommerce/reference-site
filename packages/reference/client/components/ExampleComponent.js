import { Component } from 'react'

class ExampleComponent extends Component {
  render () {
    const {
      thisProp,
      thatProp
    } = this.props

    return (
      <div>
        `The output of { thisProp } + { thatProp } is { thisProp + thatProp }`
      </div>
    )
  }
}

export default ExampleComponent
