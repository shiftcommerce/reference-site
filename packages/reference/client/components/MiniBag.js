// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'
import classNames from 'classnames'
import Cookies from 'js-cookie'

// Actions
import { readCart } from '../actions/cartActions'

// Objects
import Button from '../objects/Button'

export class MiniBag extends Component {
  componentDidMount () {
    this.props.dispatch(readCart())
  }

  renderViewYourBagLink (lineItemCount) {
    return (
      <Link href='/cart'>
        <div className='c-minibag__cart'>
          <a aria-label='View your cart'>
            <span className='c-minibag__cart-count' >
              { lineItemCount }
            </span>
          </a>
          <div className='c-minibag__cart-label--small'>
            <a>
              Basket
            </a>
          </div>
          <div className='c-minibag__cart-label--large'>
            <a>
              View Your Bag
            </a>
          </div>
        </div>
      </Link>
    )
  }

  renderCheckoutButton (lineItemCount) {
    const signedIn = Cookies.get('signedIn')

    return (
      <div className='c-minibag__checkout'>
        <Link href={(lineItemCount > 0) ? '/checkout' : ''}>
          <Button className={classNames({ 'o-button--disabled': (lineItemCount === 0) })} label="Checkout" status="primary" aria-label='Go to checkout'>
          </Button>
        </Link>
        { this.renderMyAccount(signedIn) }
        { this.renderLogout(signedIn) }
      </div>
    )
  }

  renderMyAccount (signedIn) {
    return (
      <Link href={(signedIn) ? '/account/myaccount' : '/account/login'} as='/account/myaccount' >
        <a className='o-header__myaccount'>My Account</a>
      </Link>
    )
  }

  renderLogout (signedIn) {
    if (signedIn) {
      return (
        <a href='/account/logout' className='o-header__myaccount'>
          Logout
        </a>
      )
    }
  }

  render () {
    const { cart } = this.props
    const lineItemCount = cart.totalQuantity

    return (
      <div className='c-header__minibag'>
        { this.renderViewYourBagLink(lineItemCount) }
        { this.renderCheckoutButton(lineItemCount) }
      </div>
    )
  }
}

function mapStateToProps (state) {
  const { cart } = state
  return { cart }
}

export default connect(mapStateToProps)(MiniBag)
