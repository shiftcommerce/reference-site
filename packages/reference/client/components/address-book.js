// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Action creators
import { deleteAddressBookEntry, setAddress } from '../actions/address-book-actions'
import { editForm } from '../actions/checkout-actions'

class AddressBook extends Component {
  setAddress (address) {
    const { formName, dispatch } = this.props

    dispatch(setAddress(address, formName))
  }

  handleAddressDeleted (address) {
    return this.props.dispatch(deleteAddressBookEntry(address))
  }

  createNewAddress () {
    const { formName, dispatch, onToggleCollapsed } = this.props
    onToggleCollapsed('edit', formName)
    dispatch(editForm(formName))
  }

  renderOptions (address) {
    const { id } = this.props
    const optionLabel = <a><b>{ address.meta_attributes.label.value }</b> -&nbsp; { address.first_name } { address.last_name },&nbsp;
      { address.address_line_1 },&nbsp; { address.postcode },&nbsp; { address.city },&nbsp; { address.country }
    </a>

    return (
      <div className='c-address-book__radio-container'>
        <input type='radio' name='address-book-radio' className='c-address-book__radio' onChange={() => { this.setAddress(address) }} checked={address.id === id} />
        <label htmlFor={address.id}>{ optionLabel }</label>
        <a className='c-address-book__delete-address' label='Delete' onClick={() => this.handleAddressDeleted(address)}>Delete</a>
      </div>
    )
  }

  renderEntries () {
    // Sort array of objects and return a new array
    // with the preferred address as the first object.
    // We always want to display the preferred-address as the first option
    const entries = this.props.addressBook.sort(function (a, b) {
      return b.preferred_shipping - a.preferred_shipping
    })

    return (
      <div className='c-address-book__entries'>
        { entries.map(address => {
          return (
            <div key={address.id}>
              { this.renderOptions(address) }
            </div>
          )
        }) }
      </div>
    )
  }

  render () {
    const { formName } = this.props

    return (
      <div className='c-address-book'>
        <h3>Your addresses</h3>
        { this.renderEntries() }
        <a className='c-address-book__new-address' onClick={() => this.createNewAddress(formName)}>+ Add a new address</a>
      </div>
    )
  }
}

const mapStateToProps = ({ checkout: { addressBook, shippingAddress: { id } } }) => {
  return { addressBook, id }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: dispatch
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddressBook)
