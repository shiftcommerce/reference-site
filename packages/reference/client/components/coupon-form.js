// Libraries
import { Component } from 'react'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import { connect } from 'react-redux'

// Actions
import { readCart } from '../actions/cart-actions'

// Request
import { addCartCouponRequest } from '../requests/cart-requests'

// Lib
import ApiClient from '../lib/api-client'

// Objects
import Button from '../objects/button'

export class CouponForm extends Component {
  async submitCoupon (couponCode) {
    const request = addCartCouponRequest(couponCode)
    const response = await new ApiClient().post(request.endpoint, request.body)
    if (response.status === 201) {
      return response.data
    }
    throw response.data
  }

  constructor () {
    super()
    this.handleSubmit = this.handleSubmit.bind(this)
    this.updateCart = this.updateCart.bind(this)
  }

  handleSubmit (values, { setSubmitting, setErrors }) {
    return this.submitCoupon(values.couponCode)
      .then(this.updateCart)
      .catch((error) => this.setAPIError(error, setErrors))
      .finally(() => setSubmitting(false))
  }

  updateCart () {
    this.props.dispatch(readCart({ force: true }))
  }

  setAPIError (error, setErrors) {
    setErrors({ couponCode: error.errors[0]['title'] })
  }

  validate (values) {
    let errors = {}
    if (!values.couponCode) errors.couponCode = 'Please enter a coupon code'
    return errors
  }

  renderForm () {
    return (
      <Formik
        initialValues={{ couponCode: '' }}
        validate={this.validate}
        onSubmit={this.handleSubmit}
      >
        { () => (
          <Form>
            <div className='c-coupon-form__field'>
              <Field type='text' name='couponCode' className='c-coupon-form__input' placeholder='Enter Promotion Code' />
              <Button
                aria-label='Apply Gift Code'
                containerClassName='c-coupon-form__button-container'
                className='c-coupon-form__button o-button--sml'
                label='Apply'
                status='primary' />
            </div>
            <ErrorMessage name='couponCode' component='div' className='c-coupon-form__error' />
          </Form>
        ) }
      </Formik>
    )
  }

  render () {
    return (
      <div className='c-coupon-form'>
        <div aria-label='Use gift card or rewards code' className='c-gift__wrapper'>
          <p className='c-coupon-form__title'>Promotion Code</p>
          { this.renderForm() }
        </div>
      </div>
    )
  }
}

export default connect()(CouponForm)
