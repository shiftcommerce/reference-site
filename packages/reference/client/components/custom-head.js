import Head from 'next/head'
import { Component } from 'react'

class CustomHead extends Component {
  render () {
    return <Head>
      <meta name='viewport' content='width=device-width, initial-scale=1' />
      <script src='https://js.stripe.com/v3/' />
      <link rel='icon' type='image/png' sizes='32x32' href='/static/favicon.png' />
    </Head>
  }
}

export default CustomHead
