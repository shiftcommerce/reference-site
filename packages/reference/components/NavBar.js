import { Component } from 'react'

// Objects
import NavBarOption from './NavBarOption'

class NavBar extends Component {
  renderNavOptions () {
    const navOptions = [
      { title: 'Mens', link: '/mens' },
      { title: 'Womens', link: '/womens' },
      { title: 'Boys', link: '/boys' },
      { title: 'Girls', link: '/girls' },
      { title: 'Other', link: '/other' },
      { title: 'Offers', link: '/offers' }
    ]

    return navOptions.map((navOption, index) =>
      <NavBarOption key={ index } index={ index } title={ navOption.title } link={ navOption.link } />
    )
  }

  render() {
    return <div className='o-nav' role='navigation' >
      { this.renderNavOptions() }
    </div>
  }
}

export default NavBar