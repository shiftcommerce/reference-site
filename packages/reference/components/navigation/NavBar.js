// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Objects
import NavBarOption from './NavBarOption'

class NavBar extends Component {
  renderNavOptions (categories) {
    return (
      categories.data.map((category, index) =>
        <NavBarOption key={index} index={index} title={category.title} href={`/categories/category?id=${category.id}`} as={`/categories/${category.id}`} />
      )
    )
  }

  render () {
    const {
      categories
    } = this.props

    if (categories.loading) {
      return (
        <p>Loading...</p>
      )
    } else if (categories.error) {
      return (
        <p style={{'textAlign': 'center'}} >
          Sorry! There is an error in loading menus
          {categories.error}
        </p>
      )
    } else {
      return (
        <div className='o-nav' role='navigation' >
          { this.renderNavOptions(categories) }
        </div>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    categories: state.categories || []
  }
}

export default connect(mapStateToProps)(NavBar)
