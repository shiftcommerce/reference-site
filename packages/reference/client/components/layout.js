// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'
import classNames from 'classnames'

// Objects
import { Footer, Image, Logo, NavBar, SearchBar } from 'shift-react-components'

// Components
import MiniBag from './mini-bag'
import CustomHead from './custom-head'

// Stylesheet
import '../scss/main.scss'

export class Layout extends Component {
  // serviceWorker () {
  //  // Install service worker only in production environment
  //   if (process.env.NODE_ENV === 'production') {
  //    // Registration of service worker
  //     if (typeof navigator !== 'undefined' && 'serviceWorker' in navigator) {
  //       navigator.serviceWorker.getRegistration('/app').then(registration => {
  //        // Check if service worker has already registered
  //        // register only if not yet done
  //         if (registration === undefined) {
  //           navigator.serviceWorker.register('/serviceWorker.js', { scope: '/' }).then(registration => {
  //            // Successfully registered the Service Worker
  //             console.log('Service Worker registration successful with scope: ', registration.scope)
  //           }).catch(err => {
  //            // Failed to register the Service Worker
  //             console.log('Service Worker registration failed: ', err)
  //           })
  //         }
  //       })
  //     } else {
  //       console.log('Service workers are not supported.')
  //     }
  //   }
  // }
  constructor (props) {
    super(props)
    this.state = {
      shrunk: false,
      toggleShowClass: false
    }

    this.handleScroll = this.handleScroll.bind(this)
    this.toggleDropDown = this.toggleDropDown.bind(this)
  }

  componentDidMount () {
    window.addEventListener('scroll', this.handleScroll)
  }

  componentWillUnmount () {
    window.removeEventListener('scroll', this.handleScroll)
  }

  handleScroll () {
    const scrollTop = document.documentElement.scrollTop || document.body.scrollTop
    if (scrollTop > 40 && !this.state.shrunk) {
      this.setState(state => {
        return { shrunk: true }
      })
    } else if (scrollTop <= 40 && this.state.shrunk) {
      this.setState(state => {
        return { shrunk: false }
      })
    }
  }

  renderNav () {
    return (
      <span className='o-nav u-visible-d'>
        <NavBar menu={this.props.menu} />
      </span>
    )
  }

  renderMobileNav () {
    return (
      <span className='o-nav u-hidden-d'>
        <NavBar menu={this.props.menu} />
      </span>
    )
  }

  toggleDropDown () {
    const toggleShow = this.state.toggleShowClass
    this.setState({ toggleShowClass: !toggleShow })
  }

  renderHeaderAccount (loggedIn) {
    const signedIn = loggedIn ? 'My Account' : <Link href='/account/login'><a>Sign In</a></Link>

    return (
      <div className='c-header__account' onClick={ this.toggleDropDown } >
        <Image className='c-header__account-image' src='/static/account-icon.svg' />
        { signedIn }
        { this.renderAccountDropDown() }
      </div>
    )
  }

  renderAccountDropDown () {
    const showClass = this.state.toggleShowClass
    const { loggedIn } = this.props

    const addShowClass = showClass ? 'show' : ''

    if (loggedIn) {
      return (
        <div className={classNames('c-header__dropdown-wrapper', addShowClass)} >
          <div className={classNames('c-header__account-dropdown', addShowClass)} >
            <div className='c-header__callout' />
            <Link href='/account/myaccount'>Order History</Link>
            <Link href='/account/forgotpassword'>Change Password</Link>
            <Link href='/account/logout'>Sign Out</Link>
          </div>
        </div>
      )
    } else {
      return null
    }
  }

  renderHeader () {
    const { loggedIn } = this.props
    const logoSrc = '../static/shopgo-logo.svg'

    if (typeof window === 'undefined' || window.location.pathname !== '/checkout') {
      const headerClasses = classNames('o-header', {
        'o-header--shrunk': this.state.shrunk
      })

      return (
        <>
          <CustomHead />
          <div className={ headerClasses }>
            <div className='o-header__top'>
              <div className='o-header__top-wrapper'>
                <Logo className='o-header__logo' logoSrc={logoSrc} />
                { this.renderMobileNav() }
                { this.renderHeaderAccount(loggedIn) }
                <MiniBag />
                { this.renderSearch() }
              </div>
            </div>
            { this.renderNav() }
          </div>
        </>
      )
    }
  }

  renderSearch () {
    return (
      <span className='c-header__search'>
        <SearchBar query={this.props.query} />
      </span>
    )
  }

  render () {
    const notCheckout = /^(?!\/checkout).*$/
    const bodyClassApplied = (typeof window === 'undefined' || notCheckout.test(window.location.pathname))

    const bodyClasses = classNames({
      'o-body': bodyClassApplied
    })

    return (
      <>
        { this.renderHeader() }
        <div className={bodyClasses}>
          { this.props.children }
        </div>
        <div className='o-footer'>
          <Footer />
        </div>
      </>
    )
  }
}

function mapStateToProps (state) {
  const { login, menu } = state

  return { login, menu }
}

export default connect(mapStateToProps)(Layout)
