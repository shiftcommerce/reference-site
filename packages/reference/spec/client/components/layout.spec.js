// Components
import { Layout } from '../../../client/components/layout'
import NavBar from '../../../client/components/navigation/navbar'
import MiniBag from '../../../client/components/mini-bag'

// Objects
import { Logo } from 'shift-react-components'

test('renders the header', () => {
  // Arrange

  // Act
  const wrapper = shallow(
    <Layout />
  )

  const header = wrapper.find('div.o-header')

  // Assert
  expect(header).toExist()
  expect(header).toContainReact(<Logo className="o-header__logo" logoSrc="../static/shopgo-logo.svg" />)
  expect(header.find(MiniBag)).toExist()
  expect(wrapper).toMatchSnapshot()
})

test('renders the navbar', () => {
  // Arrange

  // Act
  const wrapper = shallow(
    <Layout />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find(NavBar)).toExist()
})

test('renders the footer', () => {
  // Arrange

  // Act
  const wrapper = shallow(
    <Layout />
  )

  // Assert
  expect(wrapper.find('div.o-footer')).toIncludeText('Footer')
  expect(wrapper).toMatchSnapshot()
})

test('renders the children inside it', () => {
  // Arrange

  // Act
  const wrapper = shallow(
    <Layout>
      Content
    </Layout>
  )

  // assert
  expect(wrapper.find('div.o-body')).toIncludeText('Content')
  expect(wrapper).toMatchSnapshot()
})
