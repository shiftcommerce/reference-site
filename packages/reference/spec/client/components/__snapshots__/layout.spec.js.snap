// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the children inside it 1`] = `
<React.Fragment>
  <React.Fragment>
    <CustomHead />
    <div
      className="o-header"
    >
      <Logo
        className="o-header__logo"
      />
      <div
        className="c-header__account"
      >
        <div
          className="c-header__account-button"
        >
          <div
            className="c-header__account-label"
          >
            <Link
              href="/account/login"
            >
              <a>
                Account
              </a>
            </Link>
          </div>
        </div>
      </div>
      <Connect(MiniBag) />
      <span
        className="o-nav"
      >
        <Connect(NavBar) />
      </span>
      <span
        className="c-header__search"
      >
        <SearchBar>
          Content
        </SearchBar>
      </span>
    </div>
  </React.Fragment>
  <div
    className="o-body"
  >
    Content
  </div>
  <div
    className="o-footer"
  >
    <Connect(Footer) />
  </div>
</React.Fragment>
`;

exports[`renders the footer 1`] = `
<React.Fragment>
  <React.Fragment>
    <CustomHead />
    <div
      className="o-header"
    >
      <Logo
        className="o-header__logo"
      />
      <div
        className="c-header__account"
      >
        <div
          className="c-header__account-button"
        >
          <div
            className="c-header__account-label"
          >
            <Link
              href="/account/login"
            >
              <a>
                Account
              </a>
            </Link>
          </div>
        </div>
      </div>
      <Connect(MiniBag) />
      <span
        className="o-nav"
      >
        <Connect(NavBar) />
      </span>
      <span
        className="c-header__search"
      >
        <SearchBar />
      </span>
    </div>
  </React.Fragment>
  <div
    className="o-body"
  />
  <div
    className="o-footer"
  >
    <Connect(Footer) />
  </div>
</React.Fragment>
`;

exports[`renders the header 1`] = `
<React.Fragment>
  <React.Fragment>
    <CustomHead />
    <div
      className="o-header"
    >
      <Logo
        className="o-header__logo"
      />
      <div
        className="c-header__account"
      >
        <div
          className="c-header__account-button"
        >
          <div
            className="c-header__account-label"
          >
            <Link
              href="/account/login"
            >
              <a>
                Account
              </a>
            </Link>
          </div>
        </div>
      </div>
      <Connect(MiniBag) />
      <span
        className="o-nav"
      >
        <Connect(NavBar) />
      </span>
      <span
        className="c-header__search"
      >
        <SearchBar />
      </span>
    </div>
  </React.Fragment>
  <div
    className="o-body"
  />
  <div
    className="o-footer"
  >
    <Connect(Footer) />
  </div>
</React.Fragment>
`;

exports[`renders the navbar 1`] = `
<React.Fragment>
  <React.Fragment>
    <CustomHead />
    <div
      className="o-header"
    >
      <Logo
        className="o-header__logo"
      />
      <div
        className="c-header__account"
      >
        <div
          className="c-header__account-button"
        >
          <div
            className="c-header__account-label"
          >
            <Link
              href="/account/login"
            >
              <a>
                Account
              </a>
            </Link>
          </div>
        </div>
      </div>
      <Connect(MiniBag) />
      <span
        className="o-nav"
      >
        <Connect(NavBar) />
      </span>
      <span
        className="c-header__search"
      >
        <SearchBar />
      </span>
    </div>
  </React.Fragment>
  <div
    className="o-body"
  />
  <div
    className="o-footer"
  >
    <Connect(Footer) />
  </div>
</React.Fragment>
`;
